tests/test_utils.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017 "Shopify inc." All rights reserved.
+# Use of this source code is governed by a MIT-style license that can be found in the LICENSE file.
+from __future__ import unicode_literals
+
+import xml.etree.ElementTree as et
+
+import pytest
+import tests.utils
+
+
+@pytest.fixture
+def valid_workflow():
+    return """
+<workflow-app xmlns="uri:oozie:workflow:0.5" name="descriptive-name">
+    <parameters>
+        <property>
+            <name>property_key</name>
+            <value>property_value</value>
+        </property>
+        <property>
+            <name>other_property_key</name>
+            <value>🤣</value>
+        </property>
+    </parameters>
+    <global>
+        <job-tracker>job-tracker</job-tracker>
+        <name-node>name-node</name-node>
+    </global>
+    <start to="end" />
+    <end name="end" />
+</workflow-app>
+""".strip()
+
+
+def test_xml_to_dict_unordered():
+    document_dict = tests.utils.xml_to_dict_unordered("""
+<root>
+  <tag />
+  <tag key="value" />
+  <tag>Text</tag>
+</root>
+    """.strip())
+    assert document_dict == {
+        'root': {
+            'tag': [None, u'Text', {'@key': 'value'}]
```

Comment:
What does `<tag key="value">Text</tag>` look like? Does that throw an exception if it was included here?
What does `<tag different-key="different-value> Text </tag>` look like?